providers/flagd/pkg/cache/in_memory.go

```diff
@@ -0,0 +1,50 @@
+package cache
+
+import (
+	"context"
+	"sync"
+)
+
+type InMemory struct {
```

Comment:
This is a simple in memory implementation, good for testing purposes and can be used in cases where growing cache isn't a concern. In cases where cache size is a concern I'd suggest using a least recently used (LRU) caching implementation like [hashicorp's golang-lru](https://github.com/hashicorp/golang-lru).
Any thoughts against introducing this package in the next phase? @toddbaert @james-milligan @beeme1mr @AlexsJones

Reply:
1.19 brings [sync.Map](https://pkg.go.dev/sync#Map) , but I don't know if we want to upgrade just for this.

Reply:
I guess another candidate might be https://github.com/orcaman/concurrent-map